### DISCUSSION TITLE
.NET 6 and the dotnet-format tool -

### DISCUSSION THREAD

**Discussion by murphomatic** [nx-dotnet/nx-dotnet#454] [Q&A] [answered]
We've been using the nx-dotnet plugin now for the last couple of months with .NET 6, and have run into some potentially weird situations we can get into with regard to the dotnet-format tool.

For example - we may run some linting task on a .NET 6 project in our Nx workspace, which automagically installs the dotnet-format tool with an `--add-source` flag which points it to some v6.* repo.  This is all fine and good ... until we push our changes to the remote, and some other poor dev clones it and sets up his environment for the first time.

He runs through an npm install, which fires off a hook to do a `dotnet tool restore` ... which it attempts to do, but cannot find the v6 that is referenced in dotnet-tools.json because this is a dev version, and not an official release.

This begs the question:  If the dotnet-format tool was rolled into the .NET 6 SDK - why do we even need to install this tool?  We could just be running `dotnet format <project> --verify-no-changes` - which the executor seems to be able to do - but there is a forceToolUsage flag flipped to *true* which prevents this and forces the installation of the dotnet-format tool.  It's almost as if this forceToolUsage flag should only be true for .NET 5 and lower, and .NET 6 should rely on the in-built tooling of the SDK.

Thoughts?  Am I approaching / thinking about this incorrectly?

**Comment by murphomatic** [comment, marked as answer]
Appears that if I manually add the https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json source to my nuget.config file, the dotnet tool restore works fine.  But the question still remains - why not just use `dotnet format` instead of `dotnet tool run dotnet-format` for .NET 6+ projects?

**Reply by AgentEnder** [reply]
`dotnet format` (the builtin version) was broken between .NET SDK versions 6.0.0 and 6.0.203. For this reason, we had to fall back on the tool version which had been fixed until the fix was published by .NET SDK. If your SDK version is 6.0.203+ you shouldn't see the tool usage anymore, you can see the checks we do for that in this area: https://github.com/nx-dotnet/nx-dotnet/blob/master/packages/core/src/executors/format/executor.ts#L38

**Reply by murphomatic** [reply]
Ahhhh makes perfect sense.  Thank you Craigory!
